### PR TITLE
feat: add ClanPlayerEvent and PlayerResetKdrEvent

### PR DESCRIPTION
**Pull request by SrBedrock**
Added event that will be triggered when resetting the kdr and added two files to the .gitignore.

### CODE REVIEW

File: src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/PlayerResetKdrEvent.java
Comment (RoinujNosde) [inline src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/PlayerResetKdrEvent.java]: I don't see the need for those methods.

**Review by RoinujNosde — CHANGES_REQUESTED** [review]

File: src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/PlayerResetKdrEvent.java
Comment (SrBedrock) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/PlayerResetKdrEvent.java]: I'm creating a system where it creates a statement for each type of transaction and getting the transaction value would be useful. It's possible to obtain this information by directly accessing the SimpleClans configuration, so I can remove it.

Comment (RoinujNosde) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/PlayerResetKdrEvent.java]: > It's possible to obtain this information by directly accessing the SimpleClans configuration

Exactly

File: src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/PlayerResetKdrEvent.java
Comment (Tomut0) [inline src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/PlayerResetKdrEvent.java:22]: Why do you return Player If you may return ClanPlayer? 🤔

Comment (SrBedrock) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/PlayerResetKdrEvent.java:22]: I followed the pattern of other events that extend PlayerEvent.

Comment (Tomut0) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/PlayerResetKdrEvent.java:22]: I believe, SimpleClans should have unified event class, some kind of ClanPlayerEvent (as PlayerEvent).

Since there are couple of events that returns getClanPlayer()

Just a mind of possible refactor. 
It's not related to your PR.

Comment (SrBedrock) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/PlayerResetKdrEvent.java:22]: Should I make this change?

Comment (Tomut0) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/PlayerResetKdrEvent.java:22]: It would be great if you do.
This ClanPlayerEvent should also extends PlayerEvent.

Comment (SrBedrock) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/PlayerResetKdrEvent.java:22]: > I believe, SimpleClans should have unified event class, some kind of ClanPlayerEvent (as PlayerEvent).
> 
> Since there are couple of events that returns getClanPlayer()
> 
> Just a mind of possible refactor. It's not related to your PR.

After this PR is merge I can make the modifications to the other events

image

File: src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/ClanPlayerEvent.java
Comment (Tomut0) [inline src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/ClanPlayerEvent.java]: I don't really like that clanPlayer.toPlayer() can throw NPE here, but I feel there is nothing to do with it.

Comment (SrBedrock) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/ClanPlayerEvent.java]: if there is anything I need to change just let me know

**Comment by SrBedrock** [comment]
Reviewing the PR I found a "problem" that I don't know the best way to solve, the event is called after the player is charged if it is set https://github.com/RoinujNosde/SimpleClans/blob/0f693329cae81036bfe1423344ed41cf3d535f64/src/main/java/net/sacredlabyrinth/phaed/simpleclans/commands/general/GeneralCommands.java#L167-L171

File: src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/ClanPlayerEvent.java
Comment (Tomut0) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/ClanPlayerEvent.java]: @RoinujNosde What do you think?

Comment (RoinujNosde) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/ClanPlayerEvent.java]: Add Player as a constructor parameter.

Comment (SrBedrock) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/ClanPlayerEvent.java]: I will close this PR and let you guys decide how it should be implemented

Comment (Tomut0) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/ClanPlayerEvent.java]: > Add Player as a constructor parameter.

It seems, it would break extended events.
So, breaking API change.

**Comment by Tomut0** [comment]
@SrBedrock Can you allow changes to maintainers or can you do force-push before my idea with ClanPlayerEvent? ☕

**Comment by SrBedrock** [comment]
> Can you allow changes to maintainers 

@Tomut0 I can't enable this option because it comes [from an organization](https://github.com/orgs/community/discussions/5634), make your changes in a different branch and I will change the branch for this PR

**Comment by SrBedrock** [comment]
closed in favor of #370

**Pull request closed without merging**
